src/main/java/net/minecraftforge/fml/client/GuiModList.java

```diff
@@ -415,8 +423,18 @@ else if (disableable != Disableable.YES)
 
             IModGuiFactory guiFactory = FMLClientHandler.instance().getGuiFactoryFor(selectedMod);
             configModButton.visible = true;
-            configModButton.enabled = guiFactory != null && guiFactory.mainConfigGui(this) != null;
-
+            configModButton.enabled = false;
+            if(guiFactory != null)
```

Comment:
I don't like making the modder implement yet another method, but I get your point. It may improve perfomance a little bit it and makes it easier to understand for modders. I implemented this in e4b6be4.